WordPress/src/main/java/org/wordpress/android/ui/media/MediaAddFragment.java

```diff
@@ -64,10 +65,14 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     public void onAttach(Activity activity) {
         super.onAttach(activity);
 
-        try {
-            mCallback = (MediaAddFragmentCallback) activity;
-        } catch (ClassCastException e) {
-            throw new ClassCastException(activity.toString() + " must implement " + MediaAddFragmentCallback.class.getSimpleName());
+        if (activity instanceof WPMainActivity) {
+            mCallback = ((WPMainActivity) activity).getMySiteFragment();
```

Comment:
Hard-coding this to work with the My Site fragment seems a bit inelegant. And is it even necessary, since [onMediaAdded](https://github.com/wordpress-mobile/WordPress-Android/blob/feature/2394-add-media-masterbar/WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java#L223) is empty in the fragment?